tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -66,46 +117,174 @@ def __validate(self):
         self.total_cases_validated += 1
         self.has_validated = True
         logger.debug(f"Attempting to execute Cat4 validation for case: {self.current_case}.")
-        if self.program_type == "TAN" and self.section == "A" and "state_fips" in self.header:
-            num_errors += self.__validate_tanf_s1_case(num_errors)
-        elif self.program_type == "TAN" and self.section == "C" and "state_fips" in self.header:
-            num_errors += self.__validate_tanf_s2_case(num_errors)
-        elif self.program_type == "TAN" and self.section == "A" and "tribe_code" in self.header:
-            num_errors += self.__validate_tribal_tanf_s1_case(num_errors)
-        elif self.program_type == "TAN" and self.section == "C" and "tribe_code" in self.header:
-            num_errors += self.__validate_tribal_tanf_s2_case(num_errors)
-        elif self.program_type == "SSP" and self.section == "A":
-            num_errors += self.__validate_ssp_s1_case(num_errors)
-        elif self.program_type == "SSP" and self.section == "C":
-            num_errors += self.__validate_ssp_s2_case(num_errors)
+
+        if self.section == "A":
```

Comment:
not here specifically, the `header`'s `section_type` value should be consistent across the program types. where i needed to make a change was initializing the `CaseConsistencyValidator` in `parse_datafile`. We have this line after originally having parsed the header initialized the `CaseConsistencyValidator`:
```python
program_type = f"Tribal {header['program_type']}" if is_tribal else header['program_type']
```

This is a bit of a side-effect that changes the `program_type` - we pass this modified version along and use it throughout the parsing code but never actually update `header['program_type']`, which was used to initialize the `CaseConsistencyValidator`. As a result, when doing cat 4 validation, we were pulling `TAN` models instead of `Tribal TAN` models (finding none, and resulting in no cat4 errors).

I moved the initialization of `CaseConsistencyValidator` to be after the modification of `program_type` and updated it to use the modified value instead of the value written to the header. I'm open to suggestions from the team on whether we should write the modified value to `header['program_type']` or otherwise mitigate the duplication/side-effects @andrew-jameson @elipe17 @raftmsohani